packages/p2p-v2/src/pages/my-ads/components/AdFormInput/AdFormInput.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { ComponentProps } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+import { getValidationRules } from '@/utils';
+import { Input, Text } from '@deriv-com/ui';
+
+type TAdFormInputProps = ComponentProps<typeof Input> & {
+    currency?: string;
+    label: string;
+    name: string;
+};
+
+const RightPlaceholder = ({ value = '' }: { value?: string }) => <Text color='less-prominent'>{value}</Text>;
+
+const AdFormInput = ({ currency, label, name, ...props }: TAdFormInputProps) => {
```

Comment:
since we are extending the input here, maybe we can pass the rightPlaceholder as a prop instead of passing the currency?